[core] [nextjs] [templates/nextjs] Streamline site resolution accross sitecore client and middlewares (#141)
Sitecore/content-sdk: CHANGELOG.md

  * `NEXT_PUBLIC_SITECORE_SITE_NAME` → `NEXT_PUBLIC_DEFAULT_SITE_NAME`
  * `DISABLE_SSG_FETCH` → `GENERATE_STATIC_PATHS`
  * `disableStaticPaths` config property → `generateStaticPaths` (with inverted logic for clarity)

> * `[core]` `[nextjs]` `[templates/nextjs]` Refactor site resolution logic across packages ([#141](https://github.com/Sitecore/content-sdk/pull/141))

Reviewer: Add:
* Removed support for passing a custom `siteResolver` to `SitecoreClient`

>   * Removed `sites` paramter from `SitecoreClientInit` type

Reviewer 2: Just a small typo here

>   * Removed `sites` paramter from `SitecoreClientInit` type

Reviewer: ```suggestion
  * Removed `sites` parameter from `SitecoreClientInit` type
```

  * Removed `SiteResolver` dependency and `resolveSite()` from `SitecoreClient`

>   * Renamed Page type's `site` to `siteName`, which now holds only the site name

Reviewer: ```suggestion
  * Updated `SitecoreClient` to construct the `Page` using `siteName` instead of the full `SiteInfo`.
```

>   * Updated SitecoreClient's `getPagePaths()` to accept a sites parameter

Reviewer: ```suggestion
  * Updated SitecoreClient's `getPagePaths()` to accept a `sites` parameter
```

>   * Modified site paths service to receive `sites` in fetchSiteRoutes() instead of via constructor

Reviewer: ```suggestion
  * Modified the `getPagePaths` method in `SitecoreClient` to accept a `sites` parameter.
```

>   * Adjusted Next.js API middlewares (sitemap, robots) to use their own SiteResolver

Reviewer: ```suggestion
  * Updated Next.js `SitemapMiddleware` and `RobotsMiddleware` to use their own instance of `SiteResolver` and accept a `sites` parameter via the constructor.
```

>   * Synced Next.js template with the above changes

Reviewer: Can be removed, this is expected by default
```suggestion
  * Synced Next.js template with the above changes
```
